plugins/node/opentelemetry-instrumentation-bunyan/src/types.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span } from '@opentelemetry/api';
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type LogHookFunction = (record: Record<string, any>, span: Span) => void;
```

Comment:
In many instrumentations the "hook" function signature's first parameter is `span`, and then the other specific datarelated to the hook:
[HTTP](https://github.com/open-telemetry/opentelemetry-js/blob/9371029ee22b2a28c118a80173c2a8990f90255a/packages/opentelemetry-instrumentation-http/src/types.ts#L54)
[ioredis](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/1381a9a0f9e45a7dae53239489c68b0de95e2819/plugins/node/opentelemetry-instrumentation-ioredis/src/types.ts#L52)
and all the instrumentation [here](https://github.com/aspecto-io/opentelemetry-ext-js)

Might be worth keeping this structure if you find it appropriate here as well